### ISSUE TITLE
Bug Report | Deployment stops working after some time

### ISSUE THREAD

**Issue by pred2k** [spikecodes/libreddit#149]
**Describe the bug**
From time to time after intensive libreddit usage, my instance stops working until i restart the container.

Container Logs:
```
libreddit    | thread 'blocking-87' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:210:18
libreddit    | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
libreddit    | thread 'blocking-81' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'blocking-82' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'blocking-83' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:182:43
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'blocking-84' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'blocking-85' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'blocking-86' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:160:43
libreddit    | thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:182:43
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:182:43
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:182:43
libreddit    | thread 'async-std/runtime' panicked at 'task has failed', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
libreddit    | thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/blocking-1.0.2/src/lib.rs:182:43
```

```
docker ps -q | xargs  docker stats --no-stream
CONTAINER ID   NAME                          CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
57e764721607   libreddit                     0.00%     44MiB / 128MiB        34.38%    158MB / 132MB     0B / 0B           3
```

**To reproduce**
1. Deploy **v0.4.2** container
2. Surf trough **hot** content and play videos
3. It begins with not loading images
4. See error logs

**Expected behavior**
A non functional instance could be detected by a [container healthcheck](https://docs.docker.com/engine/reference/builder/#healthcheck) and should be restarted automatically.

**Additional context**
My [docker-compose.yaml](https://github.com/pred2k/docker-deployments/blob/main/libreddit/docker-compose.yaml) has some strict hardening directives, like `read_only`, `mem_limit`, ...

**Comment by spikecodes** [comment]
Thanks for this bug report, the healthcheck is a good idea! This issue might be coming from `async-std` but I'm working on a rewrite of Libreddit using Hyper which might fix this. I'll let you know once I've released this rewritten version.

**Comment by spikecodes** [comment]
I've just released [v0.5](https://github.com/spikecodes/libreddit/releases/v0.5.0) which should fix this. Let me know if the error persists!

**Comment by pred2k** [comment]
surfed around a while without any problems.

Thanks for the great work!
